control-plane/go.mod

```diff
@@ -11,7 +11,7 @@ require (
 	github.com/hashicorp/consul-k8s/control-plane/cni v0.0.0-20230825213844-4ea04860c5ed
 	github.com/hashicorp/consul-server-connection-manager v0.1.6
 	github.com/hashicorp/consul/api v1.26.1
-	github.com/hashicorp/consul/proto-public v0.5.1
+	github.com/hashicorp/consul/proto-public v0.1.2-0.20231107172454-028f1d8c502f
```

Comment:
You should be able to rebase on `main` as of this morning to use a replace directive for this. I handled it in #3194. We want to use a replace directive instead of swapping this value because `api` depends on `proto-public` `v0.5.1` and you can run into some dependency errors if you just swap the values in place.